### ISSUE TITLE
unable to run assemblePartial on 10x datasets

### ISSUE THREAD

**Issue by michaeleekk** [milaboratory/mixcr#706]
## Checklist before submitting the issue:

 - [x] The issue is strongly related to the MiXCR software
 - [x] The issue can be reproduced with [the most recent version](https://github.com/milaboratory/mixcr/releases) of MiXCR
 - [x] There is no answer to the question in the [official documentation](http://mixcr.readthedocs.io/) and there is no duplicate issue in the [bug tracker](https://github.com/milaboratory/mixcr/issues?utf8=✓&q=)  
 - [ ] Inspection of raw alignments with [exportAlignmentsPretty](http://mixcr.readthedocs.io/en/master/export.html#exporting-well-formatted-alignments-for-manual-inspection) shows that data has the expected architecture, and sample preparation artefacts are not the reason of the problem (if this is the matter of the issue)

## Expected Result

Able to run `assemblePartial` on 10x dataset suggested in [this page](https://github.com/milaboratory/mixcr/wiki/Analysis-of-10X-Single-Cell-data).

## Actual Result

Got an error,

```
Running assemble partial: 0%
Running assemble partial: 7.5%  ETA: 00:24:41
picocli.CommandLine$ExecutionException: Error while running command (com.milaboratory.mixcr.cli.CommandAssemblePartialAlignments@69afa141): java.lang.RuntimeException
        at picocli.CommandLine.executeUserObject(CommandLine.java:1778)
        at picocli.CommandLine.access$900(CommandLine.java:145)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2141)
        at com.milaboratory.mixcr.cli.Main$1.handle(Main.java:140)
        at com.milaboratory.mixcr.cli.Main$1.handle(Main.java:129)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1968)
        at com.milaboratory.mixcr.cli.Main.handleParseResult(Main.java:151)
        at com.milaboratory.mixcr.cli.Main.main(Main.java:123)
Caused by: java.lang.RuntimeException
        at com.milaboratory.mitool.helpers.PipeKt$group$1.take(Pipe.kt:52)
        at com.milaboratory.mitool.helpers.PipeKt$group$1.take(Pipe.kt:44)
        at cc.redberry.pipe.CUtils$OPIterator.hasNext(CUtils.java:484)
        at com.milaboratory.mixcr.cli.CommandAssemblePartialAlignments.run1(CommandAssemblePartialAlignments.java:141)
        at com.milaboratory.cli.ACommandWithSmartOverwrite.run0(ACommandWithSmartOverwrite.java:118)
        at com.milaboratory.cli.ACommand.run(ACommand.java:100)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1769)
        ... 7 more
```

## Exact MiXCR commands 

I followed [this page](https://github.com/milaboratory/mixcr/wiki/Analysis-of-10X-Single-Cell-data) and used `rna-seq` for the align step,

```bash
# alignment
java -jar /opt/bin/mixcr.jar align \
    --tag-pattern '^(CELL:N{16})(UMI:N{10})\^(R2:*)' \
    -p rna-seq -s hs \
    -OvParameters.geneFeatureToAlign=VTranscript \
    -OvParameters.parameters.floatingLeftBound=false \
    -OjParameters.parameters.floatingRightBound=false \
    -OcParameters.parameters.floatingRightBound=false \
    -OallowPartialAlignments=true \
    -OallowNoCDR3PartAlignments=true \
    --report align.report \
    $FASTQ_R1 \
    $FASTQ_R2 \
    alignments.vdjca

# Correction of barcode sequences
java -jar /opt/bin/mixcr.jar correctAndSortTags --report barcode.correction.report alignments.vdjca alignments.corrected.vdjca

# Partial assembly of CDR3 sequences from halves
$MIXCR assemblePartial -f alignments.corrected.vdjca alignments.partial.assemble.vdjca

```

## MiXCR report files 

Attached the align.report,

```
Analysis date: Wed Jun 15 23:43:35 JST 2022
Input file(s): /data/datasets/22JU002/220421_3839-6-1_22JU002_LIBRAseq_scBCR_S2_L001_R1_001.fastq.gz,/data/datasets/22JU002/220421_3839-6-1_22JU002_LIBRAseq_scBCR_S2_L001_R2_001.fastq.gz
Output file(s): alignments.vdjca
Version: 4.0.0; built=Fri Jun 10 10:07:13 JST 2022; rev=c604f8d43b; lib=repseqio.v2.0
Command line arguments: align --tag-pattern ^(CELL:N{16})(UMI:N{10})\^(R2:*) -p rna-seq -s hs -OvParameters.geneFeatureToAlign=VTranscript -OvParameters.parameters.floatingLeftBound=false -OjParameters.pa
rameters.floatingRightBound=false -OcParameters.parameters.floatingRightBound=false -OallowPartialAlignments=true -OallowNoCDR3PartAlignments=true --report align.report /data/datasets/22JU002/220421_3839-
6-1_22JU002_LIBRAseq_scBCR_S2_L001_R1_001.fastq.gz /data/datasets/22JU002/220421_3839-6-1_22JU002_LIBRAseq_scBCR_S2_L001_R2_001.fastq.gz alignments.vdjca
Analysis time: 283.15m
Total sequencing reads: 213456526
Successfully aligned reads: 157373128 (73.73%)
Alignment failed, no hits (not TCR/IG?): 53661937 (25.14%)
Alignment failed because of low total score: 2421461 (1.13%)
Overlapped: 0 (0%)
Overlapped and aligned: 0 (0%)
Alignment-aided overlaps: 0 (NaN%)
Overlapped and not aligned: 0 (0%)
No CDR3 parts alignments, percent of successfully aligned: 134745357 (85.62%)
Partial aligned reads, percent of successfully aligned: 14075613 (8.94%)
 chains: 21 (0%)
TRA chains: 43 (0%)
TRB chains: 277 (0%)
TRD chains: 2 (0%)
TRG chains: 2961 (0%)
IGH chains: 41354893 (26.28%)
TRA,TRD chains: 359 (0%)
IGK chains: 82131841 (52.19%)
IGL chains: 33882731 (21.53%)
Realigned with forced non-floating bound: 0 (0%)
Realigned with forced non-floating right bound in left read: 0 (0%)
Realigned with forced non-floating left bound in right read: 0 (0%)
Total reads: 213456526
Matched reads: 213456526 (100%)
Projection +R1 +R2: 213456526 (100%)
Variants:
  0: + 213456526 (100%) = 213456526 (100%)
Cost:
  0: + 213456526 (100%) = 213456526 (100%)
CELL length:
  16: + 213456526 (100%) = 213456526 (100%)
UMI length:
  10: + 213456526 (100%) = 213456526 (100%)
R2 length:
  100: + 213456526 (100%) = 213456526 (100%)
For variant 0:
  Variants:
    0: + 213456526 (100%) = 213456526 (100%)
  Cost:
    0: + 213456526 (100%) = 213456526 (100%)
  CELL length:
    16: + 213456526 (100%) = 213456526 (100%)
  UMI length:
    10: + 213456526 (100%) = 213456526 (100%)
  R2 length:
    100: + 213456526 (100%) = 213456526 (100%)
  For projection [1, 2]:
    CELL:Left position:
      0: + 213456526 (100%) = 213456526 (100%)
    UMI:Left position:
      16: + 213456526 (100%) = 213456526 (100%)
    CELL:Right position:
      16: + 213456526 (100%) = 213456526 (100%)
    UMI:Right position:
      26: + 213456526 (100%) = 213456526 (100%)
    R2:Left position:
      0: + 213456526 (100%) = 213456526 (100%)
    R2:Right position:
      100: + 213456526 (100%) = 213456526 (100%)
    Variants:
      0: + 213456526 (100%) = 213456526 (100%)
    Cost:
      0: + 213456526 (100%) = 213456526 (100%)
    CELL length:
      16: + 213456526 (100%) = 213456526 (100%)
    UMI length:
      10: + 213456526 (100%) = 213456526 (100%)
    R2 length:
      100: + 213456526 (100%) = 213456526 (100%)
    For sub-read 0:
      CELL:Left position:
        0: + 213456526 (100%) = 213456526 (100%)
      UMI:Left position:
        16: + 213456526 (100%) = 213456526 (100%)
      CELL:Right position:
        16: + 213456526 (100%) = 213456526 (100%)
      UMI:Right position:
        26: + 213456526 (100%) = 213456526 (100%)
      Variants:
        0: + 213456526 (100%) = 213456526 (100%)
      Cost:
        0: + 213456526 (100%) = 213456526 (100%)
      CELL length:
        16: + 213456526 (100%) = 213456526 (100%)
      UMI length:
        10: + 213456526 (100%) = 213456526 (100%)
      For variant 0:
        CELL:Left position:
      CELL:Right position:
        16: + 213456526 (100%) = 213456526 (100%)
      UMI:Right position:
        26: + 213456526 (100%) = 213456526 (100%)
      Variants:
        0: + 213456526 (100%) = 213456526 (100%)
      Cost:
        0: + 213456526 (100%) = 213456526 (100%)
      CELL length:
        16: + 213456526 (100%) = 213456526 (100%)
      UMI length:
        10: + 213456526 (100%) = 213456526 (100%)
      For variant 0:
        CELL:Left position:
          0: + 213456526 (100%) = 213456526 (100%)
        UMI:Left position:
          16: + 213456526 (100%) = 213456526 (100%)
        CELL:Right position:
          16: + 213456526 (100%) = 213456526 (100%)
        UMI:Right position:
          26: + 213456526 (100%) = 213456526 (100%)
        Cost:
          0: + 213456526 (100%) = 213456526 (100%)
        CELL length:
          16: + 213456526 (100%) = 213456526 (100%)
        UMI length:
          10: + 213456526 (100%) = 213456526 (100%)
    For sub-read 1:
      R2:Left position:
        0: + 213456526 (100%) = 213456526 (100%)
      R2:Right position:
        100: + 213456526 (100%) = 213456526 (100%)
      Variants:
        0: + 213456526 (100%) = 213456526 (100%)
      Cost:
        0: + 213456526 (100%) = 213456526 (100%)
      R2 length:
        100: + 213456526 (100%) = 213456526 (100%)
      For variant 0:
        R2:Left position:
          0: + 213456526 (100%) = 213456526 (100%)
        R2:Right position:
          100: + 213456526 (100%) = 213456526 (100%)
        Cost:
          0: + 213456526 (100%) = 213456526 (100%)
        R2 length:
          100: + 213456526 (100%) = 213456526 (100%)
```

**Comment by PoslavskySV** [comment]
Hi Michael,

thank you for reporting, we are already digging into this as well as in your previous issue.

**Comment by xiaobai-95** [comment]
I met the same problem today，would you please provide me instructions how to fix this？

**Comment by PoslavskySV** [comment]
Fixed in v4.1
